packages/block-editor/src/components/rich-text/index.native.js

```diff
@@ -619,7 +628,9 @@ function RichTextWrapper(
 			deleteEnter={ deleteEnter }
 			placeholderTextColor={ placeholderTextColor }
 			textAlign={ textAlign }
-			selectionColor={ selectionColor }
+			selectionColor={
+				selectionColor ? selectionColor : defaultSelectionColor
+			}
```

Comment:
@geriux I was trying to make updating the cursor/caret/selection color as simple and non-disruptive as possible. Since `selectionColor` is usually `undefined` for most components using RichText, I thought perhaps that adding a default selection color to the base RichText component would work best. Then, whenever `selectionColor` is passed as a prop, the selectionColor can still change, and will fallback to the default color if undefined. The objective was simplicity and reducing the chance of regressions, but I'm curious if you have any feedback here (especially before I update the [failing snapshots](https://github.com/WordPress/gutenberg/actions/runs/5451346680/jobs/9917492691?pr=52269) where selectionColor was updated).

Reply:
This approach sounds good to me, having a default value in case it is not set, just a few cases would set a custom `selectionColor`, for example, blocks like Button.

I've moved this logic to the RichText component though so that it would work for other cases like the PostTitle component. I also added a case to take into account block-based theme colors and color contrast detection, since I've encountered issues in light/dark mode using different block-based themes.

These changes were applied in https://github.com/WordPress/gutenberg/pull/52269/commits/14d5695b547d09c29f4e12590b7cf2958c20b595